Signal/src/phone/PhoneManager.m

```diff
@@ -139,7 +141,16 @@ - (void)incomingCallWithSession:(ResponderSessionDescriptor *)session {
 
 - (BOOL)hasOngoingCall
 {
-    return self.curCallController != nil;
+    if (!self.curCallController) {
+        return NO;
+    }
+
+    return [self.curCallController callState].futureTermination.isIncomplete;
```

Comment:
Heh, I just opened a PR that's very similar - I'll close it. I think your first condition is unnecessary though.